src/http/Message.cc

```diff
@@ -85,6 +85,12 @@ Http::Message::parse(const char *buf, const size_t sz, bool eof, Http::StatusCod
     // find the end of headers
     const size_t hdr_len = headersEnd(buf, sz);
 
+    if(hdr_len > Config.maxReplyHeaderSize || (hdr_len == 0 && sz > Config.maxReplyHeaderSize))
+        debugs(58, 3, "Too large reply header (" << hdr_len << " > " << Config.maxReplyHeaderSize << ")");
```

Comment:
>this may be a request rather than a reply

My understanding is that this is not true (or I am misinterpreting what you have said). If it is, I don't see the point in checking hdr_len against Config.maxReplyHeaderSize here.

Either way, I have incorporated your changes.

Reply:
>> this may be a request rather than a reply

> My understanding is that this is not true

You are probably focusing on how this _direction-agnostic_ method is used in the current master code. I do not know whether it is currently used exclusively for responses and, more importantly, I do not think we should care about (or rely upon) the current situation in this context. By design, Http::Message can be a request or a response.

Thank you for incorporating my changes.